### PR TITLE
Add refresh token binding implementation consideration and update broken links

### PR DESCRIPTION
**Pull request by tplooker**
As discussed over email, this PR drafts some text to capture the requirement around refresh token binding to the client instance not just the client. The PR also addresses some broken links.

Personally I'm still not convinced that the best way to do this binding is through the client attestation key, instead I think we could add a claim that identifies the client instance in the client key attestation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
-# OAuth2 Client Authentication for Sender Constrained Tokens
+# OAuth2 Attested Key Based Client Authentication
```
Comment (Sakurann) [inline README.md:1]:
```suggestion
# The OAuth 2.0 Attested Key Based Client Authentication
```

File: README.md [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
-This is the working area for the individual Internet-Draft, "OAuth2 Client Authentication for Sender Constrained Tokens".
+This is the working area for the individual Internet-Draft, "OAuth2 Attested Key Based Client Authentication".
```
Comment (Sakurann) [inline README.md:3]:
```suggestion
This is the working area for the individual Internet-Draft, "The OAuth 2.0 Attested Key Based Client Authentication".
```

File: draft-looker-oauth-attested-key-based-client-authentication.md [hunk 3]
```diff
@@ -241,6 +241,10 @@ The following example is the decoded header and payload of a JWT meeting the pro
 
 Implementers should be aware that the design of this authentication mechanism deliberately allows for a client instance to re-use a single Client Attestation JWT in multiple interactions/requests with an authorization server, whilst producing a fresh Client Attestation PoP JWT. Client deployments should consider this when determining the validity period for issued Client Attestation JWTs as this ultimately controls how long a client instance can re-use a single Client Attestation JWT.
 
+## Refresh token binding
+
+Authorization servers issuing a refresh token in response to a token request using the "urn:ietf:params:oauth:client-assertion-type:jwt-key-attestation" client authentication method MUST bind the refresh token to the client instance NOT just the client as specified in section 6 [@!RFC6749]. To prove this binding, the client instance MUST authenticate itself to the authorization server when refreshing an access token using the "urn:ietf:params:oauth:client-assertion-type:jwt-key-attestation" authentication method. The client MUST also use the same client attestation key that was used for authentication when the refresh token was issued.
```
Comment (Sakurann) [inline draft-looker-oauth-attested-key-based-client-authentication.md:246]:
```suggestion
Authorization servers issuing a refresh token in response to a token request using the "urn:ietf:params:oauth:client-assertion-type:jwt-key-attestation" client authentication method MUST bind the refresh token to the client instance, and NOT just the client as specified in section 6 [@!RFC6749]. To prove this binding, the client instance MUST authenticate itself to the authorization server when refreshing an access token using the "urn:ietf:params:oauth:client-assertion-type:jwt-key-attestation" authentication method. The client MUST also use the same client attestation key that was used for authentication when the refresh token was issued.
```

**Review by Sakurann — APPROVED** [review]
few suggestions

File: README.md (hunk 1 above)
Comment (tplooker) [reply README.md:1]: Don't think we want an article here "The" as a start of the title. Also 2.0 is problematic in a world where 2.1 will soon be a thing.

File: README.md (hunk 2 above)
Comment (tplooker) [reply README.md:3]: As above

**Pull request merged**